test(hooks): add unit tests for @asgardeo/react hooks to cover edge c…

Kavidu23 opened the pull request:
Purpose

This PR adds unit tests for @asgardeo/react hooks to cover edge cases and improve test coverage.
Hooks tested: useBranding, useBrowserUrl, useForm, useTranslation.

The tests cover:

Normal input, boundary, and invalid cases

Error handling and default behaviors

Context absence scenarios

Screenshots generated by Vitest are included but some may appear blank (no visual output for non-UI hooks).

Related Issues

#167 — Add unit test coverage for untested code paths in @asgardeo/react SDK

Related PRs

N/A

Checklist

 Followed the [CONTRIBUTING](https://github.com/asgardeo/javascript/blob/main/CONTRIBUTING.md)
 guidelines

 Unit tests provided for hooks

Security checks

 Followed secure coding standards in [WSO2 Secure Engineering Guidelines](http://wso2.com/technical-reports/wso2-secure-engineering-guidelines)

 Confirmed that this PR doesn’t commit any keys, passwords, tokens, usernames, or other secrets

Kavidu23 committed:
test(hooks): add unit tests for @asgardeo/react hooks to cover edge cases

Kavidu23 commented:
⚠️ Note:  
Running `pnpm nx run @asgardeo/react:lint` locally gives  
`TypeError: services.getTypeAtLocation is not a function` from `@typescript-eslint/no-implied-eval`.  
This seems unrelated to the new tests and is likely an ESLint + TypeScript plugin issue.

✅ Tests run locally with 27 passed and 2 existing failures (not from new tests):
- useBrowserUrl: handles relative afterSignInUrl → Invalid URL
- useForm: setValue and validate on change → expected 'Invalid email' to be undefined

brionmario commented:
Hey @Kavidu23,

Thanks for the PR.

 Couple of things:  
  - There's an error in the test run. Have you run the tests locally?
  - ⚠️ IMPORTNANT - lets not delete the PR template.
   
```
  > @asgardeo/react@0.5.31 test /home/runner/work/javascript/javascript/packages/react
  > vitest
  
  
   RUN  v3.1.3 /home/runner/work/javascript/javascript/packages/react
  
  ⎯⎯⎯⎯⎯⎯ Unhandled Errors ⎯⎯⎯⎯⎯⎯
  
  Vitest caught 1 unhandled error during the test run.
  This might cause false positive tests. Resolve unhandled errors to make sure your tests are not affected.
  
  ⎯⎯⎯⎯⎯⎯ Unhandled Error ⎯⎯⎯⎯⎯⎯⎯
  Error: browserType.launch: Executable doesn't exist at /home/runner/.cache/ms-playwright/chromium_headless_shell-1169/chrome-linux/headless_shell
  ╔═════════════════════════════════════════════════════════════════════════╗
  ║ Looks like Playwright Test or Playwright was just installed or updated. ║
  ║ Please run the following command to download new browsers:              ║
  ║                                                                         ║
  ║     pnpm exec playwright install                                        ║
  ║                                                                         ║
  ║ <3 Playwright Team                                                      ║
  ╚═════════════════════════════════════════════════════════════════════════╝
   ❯ ../../node_modules/.pnpm/@vitest+browser@3.1.3_playwright@1.52.0_vite@6.3.5_@types+node@22.15.30_jiti@2.6.0_ligh_b8bcb4faefafff31ab1b9ccbb49ee8ae/node_modules/@vitest/browser/dist/webdriver-BH7t2pDp.js:66:54
  
  ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
  
  
   Test Files   (4)
        Tests  no tests
       Errors  1 error
     Start at  05:33:59
     Duration  792ms (transform 0ms, setup 0ms, collect 0ms, tests 0ms, environment 0ms, prepare 0ms)
  
  error during close browserType.launch: Executable doesn't exist at /home/runner/.cache/ms-playwright/chromium_headless_shell-1169/chrome-linux/headless_shell
  ╔═════════════════════════════════════════════════════════════════════════╗
  ║ Looks like Playwright Test or Playwright was just installed or updated. ║
  ║ Please run the following command to download new browsers:              ║
  ║                                                                         ║
  ║     pnpm exec playwright install                                        ║
  ║                                                                         ║
  ║ <3 Playwright Team                                                      ║
  ╚═════════════════════════════════════════════════════════════════════════╝
      at /home/runner/work/javascript/javascript/node_modules/.pnpm/@vitest+browser@3.1.3_playwright@1.52.0_vite@6.3.5_@types+node@22.15.30_jiti@2.6.0_ligh_b8bcb4faefafff31ab1b9ccbb49ee8ae/node_modules/@vitest/browser/dist/webdriver-BH7t2pDp.js:66:54 {
    name: 'Error',
    type: 'Unhandled Error',
    stacks: [
      {
        method: '',
        file: '/home/runner/work/javascript/javascript/node_modules/.pnpm/@vitest+browser@3.1.3_playwright@1.52.0_vite@6.3.5_@types+node@22.15.30_jiti@2.6.0_ligh_b8bcb4faefafff31ab1b9ccbb49ee8ae/node_modules/@vitest/browser/dist/webdriver-BH7t2pDp.js',
        line: 66,
        column: 54
      }
    ]
  }
   ELIFECYCLE  Test failed. See above for more details.
   ```

Kavidu23 commented:
Local Test Verification

All new hook tests worked locally.
(Attached screenshot shows successful Vitest results.)
<img width="599" height="159" alt="test" src="https://github.com/user-attachments/assets/860caa45-8825-4d55-8622-59ec1ae8d076" />

brionmario commented on packages/react/src/__test__/__screenshots__/useBranding.test.ts/useBranding-hook-accepts-a-config-object-without-breaking-1.png:
Why do we need these files?

Kavidu23 committed:
test: switch to jsdom environment for React tests

- Replaced Playwright browser testing config with Vitest jsdom environment
- Removed unnecessary screenshots and Playwright setup
- Simplified test configuration to match other packages and ensure CI compatibility

Kavidu23 committed:
test(react): migrate to jsdom test environment

- replaced Playwright browser config with Vitest jsdom
- removed unused screenshots and Playwright dependencies
- aligned React package test setup with other modules for CI stability

brionmario commented on packages/react/vitest.config.ts:
```diff
@@ -16,15 +16,11 @@
 
 export default defineConfig({
   test: {
-    browser: {
```
```
  ║ Looks like Playwright Test or Playwright was just installed or updated. ║
  ║ Please run the following command to download new browsers:              ║
  ║                                                                         ║
  ║     pnpm exec playwright install                                        ║
  ║                                                                         ║
  ║ <3 Playwright Team     
  ```
  
IMO, the previous issue was due to not having the playright dependency. Can't you try adding that and keep playright as the provider. Coz `jsdom` will go obsolete in the future and we need to stick with playwrite.
